src/wasm.h

```diff
@@ -1137,61 +1137,62 @@ class AllocatingModule : public Module {
     assert(curr);
+    SubType* self = static_cast<SubType*>(this);
     switch (curr->_id) {
-      case Expression::Id::BlockId: return visitBlock((Block*)curr);
-      case Expression::Id::IfId: return visitIf((If*)curr);
-      case Expression::Id::LoopId: return visitLoop((Loop*)curr);
-      case Expression::Id::BreakId: return visitBreak((Break*)curr);
-      case Expression::Id::SwitchId: return visitSwitch((Switch*)curr);
-      case Expression::Id::CallId: return visitCall((Call*)curr);
-      case Expression::Id::CallImportId: return visitCallImport((CallImport*)curr);
-      case Expression::Id::CallIndirectId: return visitCallIndirect((CallIndirect*)curr);
-      case Expression::Id::GetLocalId: return visitGetLocal((GetLocal*)curr);
-      case Expression::Id::SetLocalId: return visitSetLocal((SetLocal*)curr);
-      case Expression::Id::LoadId: return visitLoad((Load*)curr);
-      case Expression::Id::StoreId: return visitStore((Store*)curr);
-      case Expression::Id::ConstId: return visitConst((Const*)curr);
-      case Expression::Id::UnaryId: return visitUnary((Unary*)curr);
-      case Expression::Id::BinaryId: return visitBinary((Binary*)curr);
-      case Expression::Id::SelectId: return visitSelect((Select*)curr);
-      case Expression::Id::HostId: return visitHost((Host*)curr);
-      case Expression::Id::NopId: return visitNop((Nop*)curr);
-      case Expression::Id::UnreachableId: return visitUnreachable((Unreachable*)curr);
+      case Expression::Id::BlockId: return self->visitBlock((Block*)curr);
+      case Expression::Id::IfId: return self->visitIf((If*)curr);
+      case Expression::Id::LoopId: return self->visitLoop((Loop*)curr);
+      case Expression::Id::BreakId: return self->visitBreak((Break*)curr);
+      case Expression::Id::SwitchId: return self->visitSwitch((Switch*)curr);
+      case Expression::Id::CallId: return self->visitCall((Call*)curr);
+      case Expression::Id::CallImportId: return self->visitCallImport((CallImport*)curr);
+      case Expression::Id::CallIndirectId: return self->visitCallIndirect((CallIndirect*)curr);
+      case Expression::Id::GetLocalId: return self->visitGetLocal((GetLocal*)curr);
+      case Expression::Id::SetLocalId: return self->visitSetLocal((SetLocal*)curr);
+      case Expression::Id::LoadId: return self->visitLoad((Load*)curr);
+      case Expression::Id::StoreId: return self->visitStore((Store*)curr);
+      case Expression::Id::ConstId: return self->visitConst((Const*)curr);
+      case Expression::Id::UnaryId: return self->visitUnary((Unary*)curr);
+      case Expression::Id::BinaryId: return self->visitBinary((Binary*)curr);
+      case Expression::Id::SelectId: return self->visitSelect((Select*)curr);
+      case Expression::Id::HostId: return self->visitHost((Host*)curr);
+      case Expression::Id::NopId: return self->visitNop((Nop*)curr);
+      case Expression::Id::UnreachableId: return self->visitUnreachable((Unreachable*)curr);
       default: {
```

Comment:
You shouldn't need the `default` case because we build with warnings that would tell you if you were missing one of the `enum` cases (as long as the type of `curr->_id` is `enum`). The compiler then assumes `curr->_id` must be one of the `enum`'s values, and implicitly the `default` is `__builtin_unreachable()`.